### DISCUSSION TITLE
Batch convert .dds to .png and overwrite/delete source

### DISCUSSION THREAD

**Discussion by gertsch4RS** [ImageMagick/ImageMagick#6512] [Help]
### ImageMagick version

7.1.1-14

### Operating system, version and so on

OSX or WIN11

### Description

Hey, I would like to batch process around 15.000 4k .dds textures to .png and overwrite the source files int the process. I couldn't find a definite answer to this - I found "ephemeral:" which was removed due to security reasons.

Could someone help me out here please?
Preferably I'd like to do the conversion on WIN, but OSX would also work.
Cheers!

**Comment by snibgo** [comment]
This command will create a png from each dds in the current directory:
```
magick mogrify -format png *.dds
```
It won't delete the dds files. Use a shell command for that.

**Reply by gertsch4RS** [reply]
Thanks for the quick reply! Works perfectly fine, can you point me to a solution how I could delete the dds files in the process?

**Comment by snibgo** [comment]
I'm not sure what you are asking. IM won't delete the dds files.

**Reply by gertsch4RS** [reply]
That's exactly the thing, I'd like to convert around 15000 4k textures from dss to png and delete the source dds in the process. So that the drive space is freed up instantly - only keeping the pngs.
